src/Controls/src/Xaml/MarkupExtensions/BindingExtension.cs

```diff
@@ -1,11 +1,11 @@
 using System;
 using System.ComponentModel;
+using System.Linq.Expressions;
 using Microsoft.Maui.Controls.Internals;
 
 namespace Microsoft.Maui.Controls.Xaml
 {
 	[ContentProperty(nameof(Path))]
-	[AcceptEmptyServiceProvider]
 	public sealed class BindingExtension : IMarkupExtension<BindingBase>
```

Comment:
If I understand this correctly, this extension now needs to accept two services when the binding isn't compiled but it won't need them when the binding is compiled.

```suggestion
	[RequireService(
		[typeof(IXamlTypeResolver),
		 typeof(IXamlDataTypeProvider)])]
	public sealed class BindingExtension : IMarkupExtension<BindingBase>
```

I think this extension has evolved so much that it is now doing too much. Would it make sense to either have a compiled version of the extension, or have a "simplification" visitor to completely remove it when it is possible to compile the binding?

Reply:
oh, right, I forgot this was merged. I thought it was still pending

Reply:
RequireService is net9 only

Reply:
but in net9